enterprise/backend/src/metabase_enterprise/semantic_search/index.clj

```diff
@@ -413,19 +413,39 @@
   [row]
   (update row :metadata decode-pgobject))
 
+(defn- filter-can-read-indexed-entity
+  "Check permissions for indexed entities by resolving to their parent model / card"
+  [indexed-entity-docs]
+  (let [model-index-ids (map #(-> % :id (str/split #":") first parse-long) indexed-entity-docs)
```

Comment:
nit: would it make sense to depup these `model-index-ids` into a set in case a bunch of the indexed entities all have the same parent model? I guess they're only used in the `t2/select` anyway, so more like a micro optimization for that `:in` clause.